### ISSUE TITLE
Align all sceneobject parameters and move to COMPAS core

### ISSUE THREAD

**Issue by Licini** [compas-dev/compas_viewer#174]
@tomvanmele since you are already making several PRs on Scene Objects, shall we make a list about all the visualization parameters for objects and consolidate them into COMPAS core? So that ideally we don't need to define any of them in viewer

**Comment by tomvanmele** [comment]
yes that would be great. there are still quite few things that could be generalised and streamlined...

**Comment by PingHsunTsai** [comment]
base on the modification [here](https://github.com/compas-dev/compas/pull/1353/files#diff-4afda368de8dd89b75ff53f857d8fa082784d4a02f4d3299dbf479f27659bb59R90).
I think it will be good to add 
```
self.highlight_color = Color.yellow()
self.selected = False

self.line_width = 1.0
self.point_size = 1.0
```

**Comment by tomvanmele** [comment]
i have the following notes:

* all visualisation parameters should be moved to `compas.scene.SceneObject`
* for default colors we should use the defaults of the visualisation context. therefore default values of `ColorAttribute` should be `None`.
* viewer and notebook viewer contexts should have a global default for color values, which is used if no specific value is provided by the user. in the case of a CAD environment, we use the default of the CAD environment.
* `surfacecolor` shoould be `facecolor`.
